src/Microsoft.AspNetCore.Authentication/Events/FailureContext.cs

```diff
@@ -9,10 +9,14 @@ namespace Microsoft.AspNetCore.Authentication
     /// <summary>
     /// Provides failure context information to handler providers.
     /// </summary>
-    public class FailureContext : BaseControlContext
+    public class FailureContext : HandleRequestContext<RemoteAuthenticationOptions>
```

Comment:
It also prevented us from doing cool stuff here: https://github.com/aspnet/Security/pull/925#issuecomment-237386986.